src/k8s/pkg/client/kubernetes/status.go

```diff
@@ -9,14 +9,14 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// WaitApiServerReady waits until the kube-apiserver becomes available.
-func (c *Client) WaitApiServerReady(ctx context.Context) error {
+// WaitKubernetesEndpointAvailable waits until the kubernetes endpoint becomes available.
+func (c *Client) WaitKubernetesEndpointAvailable(ctx context.Context) error {
 	return control.WaitUntilReady(ctx, func() (bool, error) {
 		// TODO: use the /readyz endpoint instead
 		// We want to retry if an error occurs (=API server not ready)
 		// returning the error would abort, thus checking for nil
-		_, err := c.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
-		return err == nil, nil
+		endpoints, err := c.CoreV1().Endpoints("default").List(ctx, metav1.ListOptions{})
```

Comment:
this is just listing the endpoints in the default namespace, not checking for `kubernetes` endpoints to be available.

Reply:
true, I updated the code to actually wait for the `kubernetes` endpoint.